### PR TITLE
ERC4626 wrapper tests

### PR DESCRIPTION
**Pull request by mosinnik**
in addition to #1120 for UsdPlus usage

### CODE REVIEW

**Comment by gtaschuk** [comment]
I think I'm misunderstanding something:  Since IStaticUSDPlusToken implements 4626 (https://github.com/ovnstable/ovnstable-app/blob/772843d01064c410dd92fbc54348122654ded2c5/contracts/interfaces/IStaticUsdPlusToken.sol#L8) can't we just use the generic `ERC4626Wrapping` in the relayer?

It seems since #1120 squashed commits, the last 2 commits are the only ones relevant here, so maybe worth updating the remote `master` and rebasing the last 2 commits onto it

**Comment by mosinnik** [comment]
> I think I'm misunderstanding something: Since IStaticUSDPlusToken implements 4626 (https://github.com/ovnstable/ovnstable-app/blob/772843d01064c410dd92fbc54348122654ded2c5/contracts/interfaces/IStaticUsdPlusToken.sol#L8) can't we just use the generic `ERC4626Wrapping` in the relayer?
> 
> It seems since #1120 squashed commits, the last 2 commits are the only ones relevant here, so maybe worth updating the remote `master` and rebasing the last 2 commits onto it

4626 give ability to convert USDC<->staticUsdPlus where USDC is asset, and staticUsdPlus is vault token in terms of 4626. But users should work with UsdPlus, so need another wrapper staticUsdPlus<->UsdPlus.
Same behavior in AAVE wrapper

**Comment by mosinnik** [comment]
Remove UsdPlus particular code and rework tests to 4626, also updated exchanges for 4626

File: pkg/standalone-utils/test/ERC4626Wrapping.test.ts [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+    // Authorize Relayer for all actions
+    const relayerActionIds = await Promise.all(
+      ['swap', 'batchSwap', 'joinPool', 'exitPool', 'setRelayerApproval', 'manageUserBalance'].map((action) =>
+        actionId(vault.instance, action)
+      )
+    );
+    const authorizer = await deployedAt('v2-vault/Authorizer', await vault.instance.getAuthorizer());
+    const wheres = relayerActionIds.map(() => ANY_ADDRESS);
+    await authorizer.connect(admin).grantPermissions(relayerActionIds, relayer.address, wheres);
```
Comment (TomAFrench) [inline pkg/standalone-utils/test/ERC4626Wrapping.test.ts:70]: The `Vault` helper class has a `grantPermissionsGlobally` method which will simplify this snippet.

File: pkg/standalone-utils/test/ERC4626Wrapping.test.ts [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+    // Approve relayer by sender
+    await vault.instance.connect(senderUser).setRelayerApproval(senderUser.address, relayer.address, true);
```
Comment (TomAFrench) [inline pkg/standalone-utils/test/ERC4626Wrapping.test.ts:73]:
```suggestion
    await vault.setRelayerApproval(senderUser, relayer, true);
```

Another method on the helper class we can use

File: pkg/standalone-utils/contracts/test/MockERC4626Token.sol [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        ERC20(_depositAsset).transferFrom(msg.sender, address(this), assets);
+        uint256 shares = assets.mulDown(_rateScale).divDown(_rate);
+        // need update to work with totalSupply
+        _mint(receiver, shares);
+        return shares;
+    }
+
+    function redeem(uint256 shares, address receiver, address owner) external override returns (uint256) {
+        uint256 assets = shares.mulDown(_rate).divDown(_rateScale);
+        // need update to work with totalSupply
```
Comment (TomAFrench) [inline pkg/standalone-utils/contracts/test/MockERC4626Token.sol:63]: Can you explain what you mean by this?

**Pull request closed without merging**